### ISSUE TITLE
sporadic unpredictable webkit_server failed to start (on ubuntu 14.04 (trusty) on circleci)

### ISSUE THREAD

**Issue by timdiggins** [thoughtbot/capybara-webkit#890]
We're using capybara-webkit on a rails project and it's working fine on a variety of local machines (using osx 10.11 and homebrew-installed qt5). It's on CI (circleci) that we're having episodic and unpredictable failures. 

Circleci is configured to use ubuntu 14.04 (trusty) -- an experimental feature that is about to go live (and we've had permission to talk about it). We upgraded to 14.04 from standard 12.04 because 14.04 comes with bundled qt5 and using 12.04 and installing qt5 was causing episodic failures due to apt-server timeouts and always taking precious extra minutes due to the install.

Circleci is using 3 parallel containers for the specs (specs randomly distributed across containers). About one in every 4 build fails, and when it fails it's generally only one of the containers that fails. All the capybara-webkit based specs in that container fail with:

```
     1.1) Failure/Error: Capybara::Webkit::Driver.new(app, Capybara::Webkit::Configuration.to_hash.merge(options))
          
          Capybara::Webkit::ConnectionError:
            /home/ubuntu/bws/vendor/bundle/ruby/2.3.0/gems/capybara-webkit-1.8.0/bin/webkit_server failed to start.
```

Any advice on how to dig into this further? I can create a branch to reduce the build to JUST run the js-rspecs (so I get the failure more quickly) and then keep running with ssh enabled until I get the failures, then ssh-in and then... well I'm not sure what I will test when I've got access to a failing container -- what are the best things to look for at that point?

FYI: We've upgraded rspec/capybara/capybara-webkit to latest just in case, but not helping. Project is using rails 4.1.x (not yet on 4.2) but don't think that's the issue. 
FYI our command to run the specs is:
```xvfb-run -a bundle exec rspec --format progress --format RspecJunitFormatter --out $CIRCLE_TEST_REPORTS/rspec/rspec-main.xml```

**Comment by fabersky** [comment]
I'm also having

>  Failure/Error: raise ConnectionError, "#{SERVER_PATH} failed to start."
>   Capybara::Webkit::ConnectionError:
>     /Users/skywalkerd/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/capybara-webkit-1.8.0/bin/webkit_server failed to start.

on OSX 10.11.4

Just upgraded to qt5-5.6.0

**Comment by fabersky** [comment]
This fixed it for me (temporary fix):

> $ gem uninstall capybara-webkit
> $ brew switch qt5 5.5.1_2 
> // the last installed version was 5.5.6.0, so if you don't have any previous version, first install it before switching to it
> $ brew linkapps qt5
> $ brew link --force qt5 
> $ gem install capybara-webkit
>

**Comment by timdiggins** [comment]
@fabersky I think your problem is unrelated to this issue. This issue is specifically about sporadic failure on an Ubuntu container

**Comment by mhoran** [comment]
@fabersky, it sounds like you're experiencing #885. capybara-webkit is not, and will not ever be, compatible with Qt 5.6. You'll need to use Qt 5.5 as you discovered (and is now covered in the wiki.)

@timdiggins, you may want to try running `/home/ubuntu/bws/vendor/bundle/ruby/2.3.0/gems/capybara-webkit-1.8.0/bin/webkit_server` directly from the command line to see if any errors are printed to the console. You'll need to run this via `xvfb-run`, or set the `DISPLAY` environment variable as appropriate.

**Comment by sgringwe** [comment]
I also am getting this error on circleci and when I SSH'd into the box and ran it, this is the error message received: `error while loading shared libraries: libQt5WebKitWidgets.so.5: cannot open shared object file: No such file or directory
`

**Comment by sgringwe** [comment]
This appears to have fixed it for me:
- sudo apt-get install libqt5webkit5

**Comment by timdiggins** [comment]
@sgringwe thanks for that -- I'm trying this out. Will have to see how it pans out in practice as this is sporadic. @mhoran thanks for help -- it's looks like a circleci install problem but I'll reopen this if it looks like anything to do with capabara-webkit

**Comment by timdiggins** [comment]
I've just reopened this as we still have sporadic largely unpredictable failures on circelci. It's clearly something to do with the circleci config, but I can't pin down why. 

I have managed to ssh in to the box after failure and successfully run the server start command. 

So I created a separate step in the ci that just runs one "smoke test" to try to do a single capybara webkit js feature. This generally (but not always succeeds) and the subsequent full run of rspec sometimes fails. Any failure will be with 

```
          Capybara::Webkit::ConnectionError:
            /home/ubuntu/bws/vendor/bundle/ruby/2.3.0/gems/capybara-webkit-1.11.0/bin/webkit_server failed to start.
```

However what often happens is that one but not the other fails. On the same server instance. And it is totally unpredictable which happens. However when it happens, all capybara webkit sessions seem to crash in the same way on that run. If I understand the code correctly this means that the bin/webkit_server actually ran, but then the calling thread failed to capture the right output data about the port number from it.

Strangely another project on circleci also using CapybaraWebkit never crashes like this. I've been combing through config to see how they differ... 

Any suggestions for what I might look for?

```
Capybara: 2.7.1
capybara-webkit: 1.11.0
Qt: 5.2.1
WebKit: 537.21
QtWebKit: 5.1.1
```

**Comment by timdiggins** [comment]
Having posted this, I then looked again at the circle config differences between the two proejcts and spotted that the one that never has this problem **isn't** using `xvfb-run -a` to run the specs. I've removed this and I think this has fixed it (given this is a sporadic problem I'll have to wait for confirmation on this, but I'll close this now).

TLDR; having problems with running capybara-webkit on circleci? Try **removing** `xvfb-run -a`.
